lnst/RecipeCommon/endpoints.py

```diff
@@ -81,3 +83,27 @@ def from_devices(cls, dev1: Device, dev2: Device) -> EndpointPair[Endpoint]:
             Endpoint(dev1),
             Endpoint(dev2),
         )
+
+
+def ip_endpoints(
+    config: EnrtConfiguration, *device_pairs: tuple[Device, Device]
```

Comment:
i don't really like that this `lnst.RecipeCommon.endpoints` function is importing an `lnst.Recipes.ENRT.BaseEnrtRecipe` class... the recipe common package should be a more "common" supporting library that is used in the Recipes package. this is in reverse, creating a sort of cycle... could actually easily lead to cyclical dependencies imo.

i think it would be safer:
1. move this method into the enrt package - if it's a helper method that we strictly use in the enrt package, and it depends on using classes/types from the enrt package... it should be in the enrt package
2. if it  is a more common use case method, it should use more common use case arguments/types. and so the method signature should be more like `ip_endpoints(*device_pairs: tuple[tuple[Device, list[IpAddress]], tuple[Device, list[IpAddress]]])`

Reply:
closing in favour of the older thread.